tests/test_gn_api.py

```diff
@@ -87,3 +87,16 @@ def record_callback(request, context):
         m.post('http://geonetwork/api/records', json=record_callback)
         zipdata = BytesIO(b"dummy_zip")
         init_gn.put_record_zip(zipdata)
+
+
+
+def test_search(init_gn):
+    with requests_mock.Mocker() as m:
+
+        def record_callback(request, context):
+            assert request.headers.get("accept") == "application/json"
+            assert request.headers.get('X-XSRF-TOKEN') == "dummy_xsrf"
+            return {"created": "success"}
+        m.post('http://geonetwork/api/records', json=record_callback)
```

Comment:
here the correct mock URL must be given.
In this case replace `http://geonetwork/api/records` with `http://geonetwork/api/search/records/_search`